Src/HALAL/Services/Communication/SPI/SPI.cpp

```diff
@@ -317,23 +452,75 @@ void HAL_SPI_RxCpltCallback(SPI_HandleTypeDef *hspi){
 
 void HAL_SPI_ErrorCallback(SPI_HandleTypeDef *hspi)
 {
-	if((hspi->ErrorCode & HAL_SPI_ERROR_UDR) != 0){
 
-		ErrorHandler("Underrun error, slave is not communicating fast enough");
+	if((hspi->ErrorCode & HAL_SPI_ERROR_UDR) != 0){
+		SPI::spi_recover(SPI::registered_spi_by_handler[hspi], hspi);
 	}else{
 		ErrorHandler("SPI error number %u",hspi->ErrorCode);
 	}
 
 }
 
+
+void SPI::spi_communicate_cache_data(SPI::Instance* spi, uint8_t* value_to_send, uint16_t size_to_send, uint8_t* value_to_receive, uint16_t aligned_size_to_receive){
+#ifdef STLIB_ETH
+	SCB_CleanDCache_by_Addr((uint32_t*)value_to_send, size_to_send);
+	SCB_InvalidateDCache_by_Addr(value_to_receive, aligned_size_to_receive);
+#endif
+	HAL_SPI_TransmitReceive_DMA(spi->hspi, value_to_send, value_to_receive, size_to_send);
+}
+
+
+void SPI::spi_end_cache_data_communication(uint8_t* value_to_receive, uint16_t size_to_receive){
+#ifdef STLIB_ETH
```

Comment:
as soon as the MPU branch is merged I'm guessing we will change this right?

Reply:
indeed